pkg/driverbuilder/kubernetes.go

```diff
@@ -270,16 +257,31 @@ func (bp *KubernetesBuildProcessor) copyModuleAndProbeFromPodWithUID(ctx context
 			if p.Status.Phase == corev1.PodRunning {
 				logger.WithField(falcoBuilderUIDLabel, falcoBuilderUID).Info("start downloading module and probe from pod")
 				if builder.ModuleFullPath != "" {
+					outModule, err := os.Create(build.ModuleFilePath)
+					if err != nil {
+						return err
+					}
+
 					err = copySingleFileFromPod(outModule, bp.coreV1Client, bp.clientConfig, p.Namespace, p.Name, builder.ModuleFullPath)
 					if err != nil {
 						return err
 					}
+					outModule.Close()
+					logger.Info("Kernel Module extraction successful")
 				}
 				if builder.ProbeFullPath != "" {
+					time.Sleep(15 * time.Second)
```

Comment:
Would it make sense to make this a config option?

Reply:
I came up, this morning, with a new idea to handle this bug with file doing a lock role.
Now the downloader script wait until the provided lockfile is deleted to cat the compiled driver